tfjs-backend-webgpu/src/kernels/resize_nearest_neighbor_webgpu.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {computeDispatch, flatDispatchLayout} from '../webgpu_util';
+
+import {WebGPUProgram} from './webgpu_program';
+
+export class ResizeNearestNeighborProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['x'];
+  workGroupSize: [number, number, number] = [64, 1, 1];
+  alignCorners: boolean;
+  halfPixelCenters: boolean;
+  oldHeight: number;
+  oldWidth: number;
+
+  constructor(
+      inputShape: [number, number, number, number], newHeight: number,
+      newWidth: number, alignCorners: boolean, halfPixelCenters: boolean) {
+    this.oldHeight = inputShape[1];
+    this.oldWidth = inputShape[2];
```

Comment:
Remove `this.oldHeight` and `this.oldWidth`? And line 31, 32

Reply:
Thanks for pointing this, fixed.